File: fix_images.js
const { execSync } = require('child_process');
const fs = require('fs');
const path = require('path');

// 1. Check and install Jimp (pure JS image library) programmatically
try {
  require.resolve('jimp');
  console.log('Jimp library is already available.');
} catch (e) {
  console.log('Jimp library is not installed. Installing jimp...');
  try {
    execSync('npm install jimp@0.16.13', { stdio: 'inherit' });
    console.log('Jimp installed successfully.');
  } catch (err) {
    console.error('Failed to install jimp automatically:', err.message);
    console.log('Please run "npm install jimp" manually in your terminal, then run this script again.');
    process.exit(1);
  }
}

const Jimp = require('jimp');

async function fixImages() {
  const imagesDir = path.join(__dirname, 'assets', 'images');
  if (!fs.existsSync(imagesDir)) {
    console.error('Images directory not found:', imagesDir);
    return;
  }

  const files = fs.readdirSync(imagesDir);
  console.log('\nScanning and fixing image extensions in:', imagesDir);
  
  for (const file of files) {
    if (file.toLowerCase().endsWith('.png')) {
      const filePath = path.join(imagesDir, file);
      try {
        // Jimp reads the image content (JPEG, WebP, PNG) and re-saves it based on the .png extension
        const image = await Jimp.read(filePath);
        await image.writeAsync(filePath);
        console.log(`✅ Formatted & verified: ${file}`);
      } catch (err) {
        console.error(`❌ Error fixing ${file}:`, err.message);
      }
    }
  }
  
  console.log('\n🎉 Done! All images are now true PNG files.');
}

fixImages();

File: fix_images.py
import os
import sys
import subprocess

def install_pillow():
    print("Checking Pillow (image library) installation...")
    try:
        from PIL import Image
        print("Pillow is already installed.")
    except ImportError:
        print("Pillow not found. Installing Pillow...")
        try:
            subprocess.run([sys.executable, "-m", "pip", "install", "Pillow"], check=True)
            print("Pillow installed successfully.")
        except Exception as e:
            print(f"Failed to install Pillow: {e}")
            print("Please run: pip install Pillow manually and then run this script.")
            sys.exit(1)

def fix_png_images():
    from PIL import Image
    assets_dir = os.path.join(os.path.dirname(os.path.abspath(__file__)), "assets", "images")
    if not os.path.exists(assets_dir):
        print(f"Directory not found: {assets_dir}")
        return

    print(f"\nScanning and fixing PNG files in: {assets_dir}")
    fixed_count = 0
    for file in os.listdir(assets_dir):
        if file.lower().endswith(".png"):
            path = os.path.join(assets_dir, file)
            try:
                # Open image and re-save as true PNG format
                with Image.open(path) as img:
                    img.save(path, "PNG")
                print(f"✅ Properly formatted & verified: {file}")
                fixed_count += 1
            except Exception as e:
                print(f"❌ Error processing {file}: {e}")
                
    print(f"\n🎉 Done! Verified and fixed {fixed_count} PNG files successfully.")

if __name__ == "__main__":
    install_pillow()
    fix_png_images()
